timer

```diff
@@ -0,0 +1,171 @@
+      echo "Version $latestVersion available"
+      echo -n "Do you wish to update $repositoryName [Y/n]: "
+      read -r answer
+      if [[ "$answer" == [Yy] ]]; then
+        cd ~ || { echo 'Update Failed'; exit 1; }
+        if [[ -d  ~/$repositoryName ]]; then rm -r -f $repositoryName || { echo "Permissions Error: try running the update as sudo"; exit 1; } ; fi
+        git clone "https://github.com/$githubUserName/$repositoryName" || { echo "Couldn't download latest version"; exit 1; }
+        cd $repositoryName || { echo 'Update Failed'; exit 1; }
+        git checkout "v$latestVersion" 2> /dev/null || git checkout "$latestVersion" 2> /dev/null || echo "Couldn't git checkout to stable release, updating to latest commit."
+        chmod a+x install.sh #this might be necessary in your case but wasnt in mine.
+        ./$nameOfInstallFile "update" || exit 1
+        cd ..
+        rm -r -f $repositoryName || { echo "Permissions Error: update succesfull but cannot delete temp files located at ~/$repositoryName delete this directory with sudo"; exit 1; }
+      else
+        exit 1
+      fi
+    else
+      echo "$repositoryName is already the latest version"
+    fi
+  fi
+}
+
+checkInternet()
+{
+  httpGet github.com > /dev/null 2>&1 || { echo "Error: no active internet connection" >&2; return 1; } # query github with a get request
+}
+
+usage()
+{
+  cat <<EOF
+timer
+Description: Enter a number of minutes and get a voice prompt when the minutes expire
+Usage: timer
+Examples:
+timer
+EOF
+}
+
+while getopts "uvh" opt; do
+  case "$opt" in
+    \?) echo "Invalid option: -$OPTARG" >&2
+        exit 1
+        ;;
+    h)  usage
+        exit 0
+        ;;
+    v)  echo "Version $currentVersion"
+        exit 0
+        ;;
+    u)  getConfiguredClient || exit 1
+        checkInternet || exit 1
+        update
+        exit 0
+        ;;
+    :)  echo "Option -$OPTARG requires an argument." >&2
+        exit 1
+        ;;
+  esac
+done
+
+# special set of first arguments that have a specific behavior across tools
+if [[ $# == "0" ]]; then
+  usage ## if calling the tool with no flags and args chances are you want to return usage
+  exit 0
+elif [[ $# == "1" ]]; then
+  if [[ $1 == "update" ]]; then
+    getConfiguredClient || exit 1
+    checkInternet || exit 1
+    update || exit 1
+    exit 0
+elif [[ $1 == "help" ]]; then
+    usage
+    exit 0
+  fi
+fi
+
+## The rest of the conditions and code would go here
+get_time () {
+    read -p "How long? Enter minutes: " the_time
+}
+
+check_for_spd-say () {
+if [[ $(which spd-say) == "" ]]
+then
+echo "You need to install spd-say to use this tool. This tool uses speech to alert you."
+echo "Try the command `sudo apt install spd-say`."
+exit 1
+fi
+}
+
+check_for_spd-say
+get_time
+#don't enter a non-number or a negative number
+while ! [ "$the_time" -ge 0 ] 2> /dev/null # must use single brackets
+    do
+    if [ "$the_time" -ge 0 ] 2> /dev/null # must use single brackets
```

Comment:
Definitely get change this statement and to ```-le``` and put the else condition under it.

Reply:
Oh, I see what I did with the "!". That was lazy. :-)

Reply:
After some experimenting, I think I must have done it on purpose. The test using -le or -lt doesn't seem to work with nonsense input like "sdfxb". It does work with negative numbers.